src/common/base32_monero.h

```diff
@@ -0,0 +1,136 @@
+ *  * name base32_crockford -> base32_monero
+ *  * name base32_crockford_alphabet -> base32_monero_alphabet
+ *  * base32_monero_alphabet contains updated alphabet (see spec below)
+ *  * base32_monero_base::normalized_symbol doesn't normalize 'i'/'I' to '1'
+ *  * base32_monero_base::normalized_symbol now normalizes 'z'/'Z' to '2'
+ *  * base32_monero_base::normalized_symbol now normalizes 'v'/'V' to 'u'
+ *  * base32_monero_base::normalized_symbol now normalizes 'l'/'L' to 'i', not '1'
+ *  * base32_monero_base::normalized_symbol now normalizes from uppercase to lowercase, not vice versa
+ *  * this comment ;)
+ *
+ * see API usage documentation here: https://github.com/tplgy/cppcodec#api-1
+ * see encoding scheme spec here: https://gist.github.com/tevador/50160d160d24cfc6c52ae02eb3d17024#35-base32-encoding
+*/
+
+#ifndef CPPCODEC_BASE32_MONERO
+#define CPPCODEC_BASE32_MONERO
+
+#include "cppcodec/detail/codec.hpp"
+#include "cppcodec/detail/base32.hpp"
+
+namespace cppcodec {
+
+namespace detail {
+
+static constexpr const char base32_monero_alphabet[32] = {
+    'x', 'm', 'r', 'b', 'a', 's', 'e', '3',
+    '2', 'c', 'd', 'f', 'g', 'h', 'i', 'j',
+    'k', 'n', 'p', 'q', 't', 'u', 'w', 'y',
+    '0', '1', '4', '5', '6', '7', '8', '9'
+};
+
+class base32_monero_base
+{
+public:
+    static CPPCODEC_ALWAYS_INLINE constexpr size_t alphabet_size() {
+        static_assert(sizeof(base32_monero_alphabet) == 32, "base32 alphabet must have 32 values");
+        return sizeof(base32_monero_alphabet);
+    }
+    static CPPCODEC_ALWAYS_INLINE constexpr char symbol(alphabet_index_t idx)
+    {
+        return base32_monero_alphabet[idx];
+    }
+    static CPPCODEC_ALWAYS_INLINE constexpr char normalized_symbol(char c)
+    {
+        // Hex decoding is always case-insensitive (even in RFC 4648), the question
+        // is only for encoding whether to use upper-case or lower-case letters.
+        return (c == 'O' || c == 'o') ? '0'
+            : (c == 'L' || c == 'l') ? 'i'
+            : (c == 'Z' || c == 'z') ? '2'
+            : (c == 'V' || c == 'v') ? 'u'
+            : (c >= 'A' && c <= 'Y') ? (c + ('a' - 'A'))
+            : c;
+    }
+
+    static CPPCODEC_ALWAYS_INLINE constexpr bool generates_padding() { return false; }
+    static CPPCODEC_ALWAYS_INLINE constexpr bool requires_padding() { return false; }
+    static CPPCODEC_ALWAYS_INLINE constexpr bool is_padding_symbol(char) { return false; }
+    static CPPCODEC_ALWAYS_INLINE constexpr bool is_eof_symbol(char c) { return c == '\0'; }
+
+    static CPPCODEC_ALWAYS_INLINE constexpr bool should_ignore(char c) {
+        return c == '-'; // "Hyphens (-) can be inserted into strings [for readability]."
+    }
+};
+
+class base32_monero : public base32_monero_base
+{
+public:
+    template <typename Codec> using codec_impl = stream_codec<Codec, base32_monero>;
+};
+
+} // namespace detail
+
+using base32_monero = detail::codec<detail::base32<detail::base32_monero>>;
```

Comment:
This feels like cheating compared to the other implementation - the PR is just offloading tons of code/work to `cppcodec`. I guess we implicitly trust that implementation?

Reply:
FWIW, the repo has >500 stars and >100 forks. You can call it "cheating", but it's using a well-established library for its intended purpose instead of reinventing the wheel.

Reply:
Having the library change underneath us is also planned to be caught in the "future_modifications_protection" test suite. That's assuming we decide to resync against the upstream repo, which can't happen by accident since git modules links by commit id.